api/routes/accountBalances.js

```diff
@@ -0,0 +1,100 @@
+var config   = require('../../config/api.config');
+var log      = require('../../lib/log')('api');
+var postgres = new require('../lib/db.js')(config.get('sql'));
+var request  = require('request');
+var response = require('response');
+
+var accountBalances = function (req, res, next) {
+
+  var options = prepareOptions();
+
+  log.info('ACCOUNT BALANCES:', options.account); 
+
+  postgres.getLedger(options, function(err, ledger){
+    if (err) {
+      errorResponse(err);
+    } else {
+      getBalances(ledger.ledger_index, options.account);
+    }
+  });
+
+  /**
+  * prepareOptions
+  * parse request parameters to determine query options 
+  */
+  function prepareOptions () {
+    var options = {
+      ledger_index : req.query.ledger_index,
+      ledger_hash  : req.query.ledger_hash,
+      datetime     : req.query.datetime,
+      currency     : req.query.currency,
+      counterparty : req.query.counterparty,
+      limit        : req.query.limit,
+      marker       : req.query.marker,
+      tx_return    : 'none',
+      account      : req.params.address
+    };
+
+    return options;
+  }
+
+  /**
+  * getBalances
+  * use ledger_index from getLedger api call
+  * to get balances using ripple REST 
+  */
+ function getBalances(ledger_index, account) {
+    if (!account) errorResponse({error: 'Must provide account.', code:400});
+    var url = 'https://api.ripple.com/v1/accounts/'+account+'/balances';
```

Comment:
We wanted the ability to get an account balance by date - the history API allows us to look up a ledger_index by date, which we can send via ripple rest to get the account balances at that ledger formatted for accounting purposes.